frontend/src/hooks/use-migrate-user-consent.ts

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import { useCurrentSettings } from "#/context/settings-context";
+import { handleCaptureConsent } from "#/utils/handle-capture-consent";
+
+export const useMigrateUserConsent = () => {
+  const { saveUserSettings } = useCurrentSettings();
+
+  /**
+   * Migrate user consent to the settings store on the server.
+   */
+  const migrateUserConsent = React.useCallback(
+    async (args?: { handleAnalyticsWasPresentInLocalStorage: () => void }) => {
+      const userAnalyticsConsent = localStorage.getItem("analytics-consent");
+
+      if (userAnalyticsConsent) {
+        args?.handleAnalyticsWasPresentInLocalStorage();
+
+        // transfer modal state to new key
+        localStorage.setItem("consent-form-closed", "true");
```

Comment:
I don't think we want to store the modal state in LS. The modal state should be captured by whether the backend has a value